core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -973,29 +972,22 @@ public boolean removePartition(long timestamp) {
             // what remains on disk
 
             // find out if we are removing min partition
-            setStateForTimestamp(path, timestamp, false);
             long nextMinTimestamp = minTimestamp;
             if (timestamp == txWriter.getPartitionTimestamp(0)) {
                 nextMinTimestamp = readMinTimestamp(txWriter.getPartitionTimestamp(1));
             }
+            long partitionNameTxn = txWriter.getPartitionNameTxnByPartitionTimestamp(timestamp);
             txWriter.beginPartitionSizeUpdate();
             txWriter.removeAttachedPartitions(timestamp);
             txWriter.setMinTimestamp(nextMinTimestamp);
             txWriter.finishPartitionSizeUpdate(nextMinTimestamp, txWriter.getMaxTimestamp());
             txWriter.commit(defaultCommitMode, denseSymbolMapWriters);
 
-            if (ff.exists(path.$())) {
```

Comment:
Reused O3 partition purge which accounts for open readers against the folder prior to deleting it.